backend/uclapi/roombookings/decorators.py

```diff
@@ -1,15 +1,81 @@
-from dashboard.models import App
+from dashboard.models import App, TemporaryToken
 from django.core.exceptions import ObjectDoesNotExist
+from django.http import JsonResponse
 from uclapi.settings import REDIS_UCLAPI_HOST
-from .helpers import PrettyJsonResponse as JsonResponse, how_many_seconds_until_midnight
+from .helpers import PrettyJsonResponse as JsonResponse, \
+    how_many_seconds_until_midnight
 
+import datetime
 import keen
 import re
 import redis
 
+
 def does_token_exist(view_func):
+
     def wrapped(request, *args, **kwargs):
         token = request.GET.get("token")
+        is_temp_token = None
+
+        try:
+            if token.split("-")[1] == "temp":
+                is_temp_token = True
+        except:
+            is_temp_token = False
+
+        if is_temp_token:
+            try:
+                temp_token = TemporaryToken.objects.get(
+                    api_token=token
+                )
+            except ObjectDoesNotExist:
+                response = JsonResponse({
+                    "ok": False,
+                    "error": "Invalid temporary token"
+                })
+                response.status_code = 400
+                return response
+
+            if request.path != "/roombookings/bookings":
+                temp_token.uses += 1
```

Comment:
should we increment the uses even if there is no "use" because we returned an error?
Just asking, this might totally be the right thing to do :)